packages/consent-manager/index.test.js

```diff
@@ -155,6 +170,45 @@ test('if accept all button is clicked, any open dialogs are closed and all analy
   expect(prefs.loadAll).toBe(true)
 })
 
+test('loads segment and additional services if loadAll is passed', async () => {
+  await runWithMockedImport(
+    './util/cookies',
+    {
+      loadPreferences: () => {
+        return { loadAll: true, segment: { foo: 'bar' } }
+      },
+      savePreferences: () => {},
+    },
+    (MockedConsentManager) => {
+      render(
+        <MockedConsentManager {...defaultProps} forceShow={false} version={2} />
+      )
+      const html = document.body.innerHTML
+      // script was injected
+      expect(html).toMatch(
+        /<script type="text\/javascript" src="https:\/\/artemis.hashicorp.com\/script/
+      )
+      // all services loaded, as well as custom segment
+      expect(html).toMatch(
+        /analytics\.load\("iyi06c432UL7SB1r3fQReec4bNwFyzkW", {"integrations":{"All":true,"Segment\.io":true,"foo":"bar"}}\);/
+      )
+
+      // custom service loaded
+      expect(html).toMatch(
+        /src="http:\/\/www.an-optional-url-for-a-script-to-add-to-the-page\.com"/
+      )
+
+      // conditionally loaded service not loaded
+      expect(html).not.toMatch(
+        /src="https:\/\/source-of-conditionally-loaded-script\.com"/
+      )
+
+      // custom inline script
+      expect(html).toMatch(/window\.foo = "bar"/)
+    }
+  )
+})
```

Comment:
🤔 is this a duplicate of the following test case?

Reply:
It appears so. 😶 🔪 https://github.com/hashicorp/react-components/pull/956/commits/a834bcbae2fb77c9f62b75baa5fa8ffc5411724c